### ISSUE TITLE
Can't Install grpc

### ISSUE THREAD

**Issue by krishnatejap** [grpc/grpc-node#1470]
### Problem description
Unable to install grpc@1.24.3. I tried installing older machines but that didn't solve the problem as well.

### Reproduction steps
Failed to install grpc module.

### Environment
 - Windows 10 Enterprise, 10.0.17763 and intel i7
 - Node version 10.15.3
 - npm 5.6.0 and 6.14.5
 - Regular installation (not through nvm)
 - gRPC#1.24.3 ( A dependency of https://www.npmjs.com/package/fabric-ca-client@1.4.8)
- windows-build-tools@5.2.2 installed. VsCode 2015 and 2017

### Additional context
gyp ERR! build error
gyp ERR! stack Error: `C:\Program Files (x86)\MSBuild\14.0\bin\msbuild.exe` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (C:\Users\krishnap\AppData\Roaming\npm\node_modules\npm\node_modules\node-gyp\lib\build.js:258:23)
gyp ERR! stack     at ChildProcess.emit (events.js:189:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:248:12)
gyp ERR! System Windows_NT 10.0.17763
gyp ERR! command "C:\\Program Files (x86)\\nodejs\\node.exe" "C:\\Users\\krishnap\\AppData\\Roaming\\npm\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "build" "--fallback-to-build" "--library=static_library" "--module=C:\\Users\\krishnap\\Ap
pData\\Roaming\\npm\\node_modules\\grpc\\src\\node\\extension_binary\\node-v64-win32-ia32-unknown\\grpc_node.node" "--module_name=grpc_node" "--module_path=C:\\Users\\krishnap\\AppData\\Roaming\\npm\\node_modules\\grpc\\src\\node\\extension_binary\\node-v64-win32-
ia32-unknown" "--napi_version=3" "--node_abi_napi=napi" "--napi_build_version=0" "--node_napi_label=node-v64"
gyp ERR! cwd C:\Users\krishnap\AppData\Roaming\npm\node_modules\grpc
gyp ERR! node -v v10.15.3
gyp ERR! node-gyp -v v3.6.2
gyp ERR! not ok
node-pre-gyp ERR! stack Error: Failed to execute 'C:\Program Files (x86)\nodejs\node.exe C:\Users\krishnap\AppData\Roaming\npm\node_modules\npm\node_modules\node-gyp\bin\node-gyp.js build --fallback-to-build --library=static_library --module=C:\Users\Panchalavarap
uK\AppData\Roaming\npm\node_modules\grpc\src\node\extension_binary\node-v64-win32-ia32-unknown\grpc_node.node --module_name=grpc_node --module_path=C:\Users\krishnap\AppData\Roaming\npm\node_modules\grpc\src\node\extension_binary\node-v64-win32-ia32-unknown --napi
_version=3 --node_abi_napi=napi --napi_build_version=0 --node_napi_label=node-v64' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (C:\Users\krishnap\AppData\Roaming\npm\node_modules\grpc\node_modules\node-pre-gyp\lib\util\compile.js:83:29)
node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:189:13)
node-pre-gyp ERR! stack     at maybeClose (internal/child_process.js:970:16)
node-pre-gyp ERR! System Windows_NT 10.0.17763
node-pre-gyp ERR! command "C:\\Program Files (x86)\\nodejs\\node.exe" "C:\\Users\\krishnap\\AppData\\Roaming\\npm\\node_modules\\grpc\\node_modules\\node-pre-gyp\\bin\\node-pre-gyp" "install" "--fallback-to-build" "--library=static_library"
node-pre-gyp ERR! cwd C:\Users\krishnap\AppData\Roaming\npm\node_modules\grpc
node-pre-gyp ERR! node -v v10.15.3
node-pre-gyp ERR! node-pre-gyp -v v0.15.0
node-pre-gyp ERR! not ok
Failed to execute 'C:\Program Files (x86)\nodejs\node.exe C:\Users\krishnap\AppData\Roaming\npm\node_modules\npm\node_modules\node-gyp\bin\node-gyp.js build --fallback-to-build --library=static_library --module=C:\Users\krishnap\AppData\Roaming\npm\node_mod

**Comment by murgatroid99** [comment]
Can you share the first 10 lines or so of the logs?

**Comment by krishnatejap** [comment]
Hi @murgatroid99 , I tried again but failed. My proxy is not allowing to download the file through npm install although I was able to do it separately. I tried by downloading the  tar ball from https://node-precompiled-binaries.grpc.io/grpc/v1.24.3/node-v64-win32-x64-unknown.tar.gz and setting it to "grpc_node_binary_host_mirror". But that didn't work as well - may be I was doing it wrongly.

**Comment by murgatroid99** [comment]
The following may work:

```
npm install --grpc --grpc_node_binary_host_mirror=https://storage.cloud.google.com/node-precompiled-binaries.grpc.io
```

This will try to download the file directly from the Google Cloud Storage bucket instead of going through Cloudflare. If that still doesn't work I don't really know what other options there are.

**Comment by sebysr** [comment]
@murgatroid99 i cannot solve the issue with the command you proposed. 😢

**Comment by murgatroid99** [comment]
Unfortunately, if the download is failing because `node-pre-gyp` can't traverse the proxy, I don't really know what else can help.

If you're using `grpc` directly, you may want to try `@grpc/grpc-js` instead. It has almost the same API and almost the same feature set, and it's the library that development efforts are going to be focused on.

**Comment by krishnatejap** [comment]
@murgatroid99 , I tried all the aforesaid approaches and still continue to face issues with installing grpc. I'm not using @grpc directly - have dependencies via @fabric-network/@fabric-client modules. I was able to sucessfully install @grpc/grpc-jc though. Since @grpc/grpc-js will be the focus area, I believe the teams responsible for these packages would migrate to the new module.

Thank you very much looking into the issue.
